### PR TITLE
[accelerator] add support for acceleration with apple pay

### PR DESCRIPTION
**Pull request by nymkappa**

### CODE REVIEW

File: frontend/src/app/components/accelerate-checkout/accelerate-checkout.component.ts [hunk 1]
```diff
@@ -60,6 +60,7 @@ export class AccelerateCheckout implements OnInit, OnDestroy {
   @Input() eta: ETA;
   @Input() scrollEvent: boolean;
   @Input() cashappEnabled: boolean = true;
+  @Input() applePayEnabled: boolean = false;
```
Comment (mononaut) [inline frontend/src/app/components/accelerate-checkout/accelerate-checkout.component.ts:63]: This input parameter doesn't seem to be used externally yet - do we want to enforce the same "acceleration eligibility criteria" as we do for cashapp payments?

see https://github.com/mempool/mempool/blob/82c271267a27678f71a519fb47259179944295e7/frontend/src/app/components/transaction/transaction.component.ts#L452-L454
https://github.com/mempool/mempool/blob/82c271267a27678f71a519fb47259179944295e7/frontend/src/app/components/tracker/tracker.component.ts#L756-L758

**Review by mononaut — APPROVED** [review]
utACK @ [8762cca]

I can't test end-to-end locally, but the initial checkout stages work well with mocked data, and the code looks fine.

I've added a couple of inline comments w/ small issues and questions, but otherwise it looks good to me to test on staging.

**Comment by wiz** [comment]
please rebase

File: frontend/src/app/components/accelerate-checkout/accelerate-checkout.component.ts [hunk 2]
```diff
@@ -131,6 +133,12 @@ export class AccelerateCheckout implements OnInit, OnDestroy {
     private enterpriseService: EnterpriseService,
   ) {
     this.accelerationUUID = window.crypto.randomUUID();
+
+    // Check if Apple Pay available
+    // @ts-ignore https://developer.apple.com/documentation/apple_pay_on_the_web/apple_pay_js_api/checking_for_apple_pay_availability#overview
+    if (window.ApplePaySession) {
+      this.applePayEnabled = true;
```
[the remark this answers is not in the document]
Comment (nymkappa) [reply frontend/src/app/components/accelerate-checkout/accelerate-checkout.component.ts:142]: Why would it not be ? It's just a fancy variable that can be set at initialization AFAIK.

**Pull request merged**